src/lib/detection_algorithm_manager/main.js

```diff
@@ -0,0 +1,53 @@
+//read from config to find detection algorthims and paths
```

Comment:
So, my main issue with the current architecture is that it still doesn't abstract the reviewer and exporter from the rest of the code.
I also don't think that this class should stateful (e.g. have a "current" exporter and reviewer)
What I have in mind is to have methods here that accept the parameters and then go from there. One option is to have an initExporter method that will return an object with an export method in it's completion callback, or even just return an exportFrame callback that can be used by the calling code.

IMO the reviewer can just be reduced to a single "review" function that accepts the folder path.